### PR TITLE
configure: fix --enable-debug for kernel module

### PR DESCRIPTION
**Pull request by andriytk**
# Problem Statement
In commit 46047958a `-O0` gcc option was changed to `-O2` for debug mode build of the kernel module to fix some compilation issues, but it will generate the code which would be hard to debug, defeating the purpose of the debug mode build.

# Design
Use special `-Og` option, which allows building kernel code and yet generates debug-friendly code.

Read more about `-Og` at https://lwn.net/Articles/756616/.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by upendrapatwardhan — COMMENTED** [review]
When verifying this locally, I ran into some compilation errors for uninitialized variables.

Started [PR-1487](https://github.com/Seagate/cortx-motr/pull/1487) to fix the compilation issues for uninitialized variables when compiling with `-Og` option.

**Comment by andriytk** [comment]
> When verifying this locally, I ran into some compilation errors for uninitialized variables.

Strange. I didn't see any compilation errors on my setup. What setup do you use and what errors did you see? Can you copy/paste them here, please?

**Comment by upendrapatwardhan** [comment]
> > When verifying this locally, I ran into some compilation errors for uninitialized variables.
> 
> Strange. I didn't see any compilation errors on my setup. What setup do you use and what errors did you see? Can you copy/paste them here, please?

I was testing on Rocky Linux release 8.4 (Green Obsidian) VM with 4.18.0-305.3.1.el8_4.x86_64 kernel.
I was building the code with the following command
`MAKE_OPTS=-j32 CONFIGURE_OPTS=--enable-debug ./scripts/m0 rebuild`

I saw the below error
```   CC [M]  /root/cortx/cortx-motr/addb2/sys.o
/root/cortx/cortx-motr/addb2/sys.c: In function ‘m0_addb2_sys_net_start_with’:
/root/cortx/cortx-motr/addb2/sys.c:333:30: error: ‘result’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  int                         result;
                              ^~~~~~
cc1: all warnings being treated as errors
make[3]: *** [scripts/Makefile.build:316: /root/cortx/cortx-motr/addb2/sys.o] Error 1
make[2]: *** [Makefile:1563: _module_/root/cortx/cortx-motr] Error 2
make[1]: *** [Makefile:20555: build-kernel-modules] Error 2```

After fixing the above error, I saw a couple of more similar errors.

**Comment by andriytk** [comment]
@upendrapatwardhan Looks like gcc-8.3 on RL-8.4 is stricter with the uninitialised variables.
I've fixed all the errors. Please review. Thank you!

File: layout/linear_enum.c [hunk 1]
```diff
@@ -145,17 +145,20 @@ M0_INTERNAL int m0_linear_enum_build(struct m0_layout_domain *dom,
...
 	M0_ENTRY("domain %p", dom);
+
 	rc = linear_allocate(dom, &e);
 	if (rc == 0) {
 		lin_enum = bob_of(e, struct m0_layout_linear_enum,
 			          lle_base, &linear_bob);
 		rc = linear_populate(lin_enum, attr);
-		if (rc == 0)
+		if (rc == 0) {
+			linear_invariant(lin_enum);
```
Comment (madhavemuri) [inline layout/linear_enum.c:155]: Use M0_ASSERT() here.

Comment (andriytk) [reply layout/linear_enum.c]: Oh, right. Fixed, thanks!

**Pull request merged**